apps/webapp/src/NewHeadline.tsx

```diff
@@ -0,0 +1,157 @@
+    const [replacementHeadline, setReplacementHeadline] = useState("Study Finds Correlation Between Coffee Consumption and Lower Risk of Certain Cancers");
+    // Each citation is an object: { url: string, explanation: string }
+    const [citations, setCitations] = useState<{ url: string; explanation: string }[]>([
+        { url: "", explanation: "" }
+    ]);
+
+    // Error state
+    const [fieldErrors, setFieldErrors] = useState<{
+        originalHeadline?: string;
+        replacementHeadline?: string;
+        citations?: string;
+    }>({});
+
+    // Handle citation URL change
+    const handleCitationUrlChange = (idx: number, value: string) => {
+        const newCitations = [...citations];
+        newCitations[idx] = { ...newCitations[idx], url: value };
+        // If last field is being edited and is not empty, add a new empty field
+        if (idx === citations.length - 1 && value.trim() !== "") {
+            newCitations.push({ url: "", explanation: "" });
+        }
+        // Remove trailing empty fields (but always keep at least one)
+        while (
+            newCitations.length > 1 &&
+            newCitations[newCitations.length - 1].url === "" &&
+            newCitations[newCitations.length - 2].url === ""
+        ) {
+            newCitations.pop();
+        }
+        setCitations(newCitations);
+    };
+
+    // Handle citation explanation change
+    const handleCitationExplanationChange = (idx: number, value: string) => {
+        const newCitations = [...citations];
+        newCitations[idx] = { ...newCitations[idx], explanation: value };
+        setCitations(newCitations);
+    };
+
+    // Form submit handler
+    const handleSubmit = (e: React.FormEvent) => {
+        e.preventDefault();
+        const errors: { originalHeadline?: string; replacementHeadline?: string; citations?: string } = {};
+        if (!originalHeadline.trim()) {
+            errors.originalHeadline = "Original headline is required.";
+        } else if (originalHeadline.length > MAX_HEADLINE_LENGTH) {
+            errors.originalHeadline = `Original headline must be at most ${MAX_HEADLINE_LENGTH} characters.`;
+        }
+        if (!replacementHeadline.trim()) {
+            errors.replacementHeadline = "Replacement headline is required.";
+        } else if (replacementHeadline.length > MAX_HEADLINE_LENGTH) {
+            errors.replacementHeadline = `Replacement headline must be at most ${MAX_HEADLINE_LENGTH} characters.`;
+        }
+        const hasCitation = citations.some(c => c.url.trim() !== "");
+        if (!hasCitation) {
+            errors.citations = "At least one citation URL is required.";
+        }
+        setFieldErrors(errors);
+        if (Object.keys(errors).length === 0) {
+            // Submit logic here (e.g. API call)
+            // For now, just log
+            console.log({ originalHeadline, replacementHeadline, citations });
+        }
+    };
+
+    return (
+        <Page>
```

Comment:
Do we want this component as its own page or a modal?

Reply:
At some point, I imagine the browser extension will have a modal version of this. I wanted to get this version solid and locked down before I duplicate it into the browser extension.

If the version in the web app was a modal, I'm not sure where it would appear.